src/main/java/hal/codec/Codec.java

```diff
@@ -0,0 +1,127 @@
+package hal.codec;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.StringJoiner;
+
+import hal.exceptions.ProcessingException;
+import hal.tasks.Deadline;
+import hal.tasks.Event;
+import hal.tasks.Task;
+import hal.tasks.Todo;
+
+/**
+ * The `Codec` class provides methods for encoding and decoding tasks.
+ */
+public class Codec {
+
+    private static final String DELIMITER = ", ";
+
+    /**
+     * Encodes a task into a string representation.
+     *
+     * @param task The task to be encoded.
+     * @return A string representation of the encoded task.
+     */
+    public String encode(Task task) {
+        String[] encodedTask = task.encode();
```

Comment:
Since this is a multi-valued variable, would it be better to use plural form here to emphasise that it is a collection?